ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -449,4 +462,91 @@ public EventDispatcher getEventDispatcher() {
   public void sendAccessibilityEvent(int tag, int eventType) {
     mUIImplementation.sendAccessibilityEvent(tag, eventType);
   }
+
+  @ReactMethod
+  public void takeSnapshot(String target, int tag, ReadableMap options, Promise promise) {
+    String format = options.hasKey("format") ? options.getString("format") : "png";
+    Bitmap.CompressFormat compressFormat =
+    format.equals("png") ? Bitmap.CompressFormat.PNG :
+    format.equals("jpg")||format.equals("jpeg") ? Bitmap.CompressFormat.JPEG :
+    format.equals("webm") ? Bitmap.CompressFormat.WEBP :
+    null;
+    if (compressFormat == null) {
+      throw new JSApplicationIllegalArgumentException("Unsupported image format: " + format);
+    }
+    double quality = options.hasKey("quality") ? options.getDouble("quality") : 1.0;
+    Integer width = options.hasKey("width") ? options.getInt("width") : null;
+    Integer height = options.hasKey("height") ? options.getInt("height") : null;
```

Comment:
note to myself: need to getDouble for width/height. and do a PixelRatio conversion in uiimplementation (or somewhere before the snapshot class)